pkg/v1/mutate/mutate.go

```diff
@@ -92,6 +93,26 @@ func AppendManifests(base v1.ImageIndex, adds ...IndexAddendum) v1.ImageIndex {
 	}
 }
 
+// RemoveManifests removes any descriptors that match the match.Matcher.
+func RemoveManifests(base v1.ImageIndex, matcher match.Matcher) (v1.ImageIndex, error) {
```

Comment:
Okay, fine, one last thing before I merge this. Sorry... thank you for your patience.

We _could_ defer this to be a little bit lazier (happens in `compute` instead of here) so that we don't return an error here, which would make using it a little easier (one less error check, matching `AppendManifests`).

This would allow for:

```go
	ii = mutate.AppendManifests(mutate.RemoveManifests(ii, matcher), add)
```

Not a _huge_ deal, but something I often find frustrating when using this library is that I am constantly checking for errors for things that don't _really_ need to return an error, and I wish I could go back and fix that.

Reply:
I understand what you are trying to do. If this only returns a `v1.ImageIndex` and not `(v1.ImageIndex, error)`, then you have the option to chain them.

I just don't see how. 

The source for this is:

```go
// RemoveManifests removes any descriptors that match the match.Matcher.
func RemoveManifests(base v1.ImageIndex, matcher match.Matcher) (v1.ImageIndex, error) {
	ii, err := base.IndexManifest()
	if err != nil {
		return base, err
	}

	// create a list without any existing ones
	manifests := []v1.Descriptor{}
	for _, manifest := range ii.Manifests {
		if !matcher(manifest) {
			manifests = append(manifests, manifest)
		}
	}
	return &index{
		base:     base,
		replaces: manifests,
	}, nil
}
```

The only potential error is from getting `base.IndexManifest()`, but it _could_ return an error. How would we handle it?

Reply:
> but it could return an error. How would we handle it?

Drop `index.replaces`, for an `index.removeMatcher` and set it to the `match.Matcher` we pass in, then instead of doing:

```go
        manifests := manifest.Manifests
	if len(i.replaces) > 0 {
		manifests = i.replaces
	}
```

in `index.compute`, do:

```go
	manifests := []v1.Descriptor
        for _, desc := range manifest.Manifests {
            if !i.removeMatcher(desc) {
                manifests = append(manifests, desc)
            }
        }
```

We're basically just deferring that `IndexManifest()` call until `compute()` is called.

`RemoveManifests` just becomes:

```go
// RemoveManifests removes any descriptors that match the match.Matcher.
func RemoveManifests(base v1.ImageIndex, matcher match.Matcher) v1.ImageIndex {
	return &index{
		base:     base,
		removeMatcher: matcher,
	}
}
```

Reply:
I think I got what you had in mind. I added it as another commit, so I can roll back if that is not it. Take a look and comment here. If it is, I will squash the two commits.